ui/desktop/src/ChatWindow.tsx

```diff
@@ -120,15 +134,41 @@ function ChatContent({
     }
   }, [messages]);
 
+  const scrollToBottom = (behavior: ScrollBehavior = 'smooth') => {
+    if (messagesEndRef.current) {
+      messagesEndRef.current.scrollIntoView({
+        behavior,
+        block: 'end',
+        inline: 'nearest'
+      });
+    }
+  };
+
+  // Single effect to handle all scrolling
+  useEffect(() => {
+    if (isLoading || messages.length > 0 || working === Working.Working) {
+      // Initial scroll
+      scrollToBottom(isLoading || working === Working.Working ? 'instant' : 'smooth');
+      
+      // Additional scrolls to catch dynamic content
+      [100, 300, 500].forEach(delay => {
```

Comment:
I wonder if this one is really needed? I would think messages would render at their height pretty much right away. Maybe try removing it?

Reply:
removed it and it works still :)